packages/zowe-explorer/src/trees/dataset/DatasetInit.ts

```diff
@@ -30,6 +31,12 @@ export class DatasetInit {
         return tree;
     }
 
+    public static async datasetTreeVisibilityChanged(this: void, e: vscode.TreeViewVisibilityChangeEvent): Promise<void> {
+        if (e.visible) {
+            await ProfilesUtils.promptUserWithNoConfigs();
+        }
+    }
```

Comment:
I like the idea of having a set of actions that can be performed as long as a specific tree is visible.

I know this is to fix one specific issue, but...

I'm curious if we should either do the same kind of visibility detection across the board, or in the shared init.

The only reason why I'm asking is because the `ProfilesUtils.promptUserWithNoConfigs` doesn't feel to be DS specific  😅

Reply:
This dialog is mostly to help new users who don't know that they don't have any Zowe client configurations. This means that when the Zowe Explorer panel is opened for the first time, it will prompt the user if no client configurations are detected. Since the Data Sets tree is the first to initialize, the event listener will do the prompting if no client configs were detected once the data sets tree is opened.

While this isn't specific to the data sets tree, I figured that it made sense to use that tree for the event subscription as it is the first tree that's initialized. I don't think there's a way to detect when Zowe Explorer is opened from the Side Bar - if there was, I think that would be the best approach. That's why I put the event on one of the tree views as it will initialize when the ZE icon is clicked in the Side Bar.

Reply:
Makes sense to keep using the first tree in our activations sequence.
I just hope we find this whenever we need to make other "if-visible" operations 😋